src/cpp/src/llm_pipeline.cpp

```diff
@@ -339,11 +394,354 @@ std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
 }  // namespace genai
 }  // namespace ov
 
+namespace {
+
+ov::Core get_core() {
+    static ov::Core core;
+    return core;
+}
+
+std::shared_ptr<ov::Model> add_slices_to_kvcache_inputs(const std::shared_ptr<ov::Model>& model) {
```

Comment:
@slyalin is there any profit to modify the existing model rather than creating a new one?

Reply:
Modifying an existing model is possible only if you have a full control over it. You shouldn't modify user-side models. If you cloned this model, then modifying in place is just simpler, no reason to create a new one.

Reply:
Got it, thanks!